lib/storage-client.js

```diff
@@ -0,0 +1,455 @@
+import { getChromedriverDir } from './utils';
+import _ from 'lodash';
+import requestPromise from 'request-promise';
+import request from 'request';
+import xpath from 'xpath';
+import { DOMParser } from 'xmldom';
+import B from 'bluebird';
+import path from 'path';
+import { system, fs, logger, tempDir, zip, util } from 'appium-support';
+
+
+const STORAGE_URL = 'https://chromedriver.storage.googleapis.com';
+const TIMEOUT_MS = 15000;
+const MAX_PARALLEL_DOWNLOADS = 5;
+
+const log = logger.getLogger('ChromedriverStorageClient');
+
+async function walkDir (dir) {
```

Comment:
this seems like the kind of thing that should already exist in the node std lib or if not, that we could put into appium-support/fs

Reply:
might be, I'll do it later